### PR TITLE
Adding GOMC free energy data sets

### PR DESCRIPTION
**Pull request by msoroush**
issue #33 
https://github.com/alchemistry/alchemlyb/pull/78

### CODE REVIEW

File: docs/gomc.rst [hunk 1]
```diff
@@ -0,0 +1,32 @@
+.. _gomc-datasets:
+
+===============
+GOMC datasets
+===============
+.. automodule:: alchemtest.gomc
+
+The :mod:`alchemlyb.gomc` module features datasets generated using the
+`GOMC <http://gomc.eng.wayne.edu/>`_ Monte Carlo simulaiton engine.  They
```
Comment (orbeckst) [inline docs/gomc.rst]: simulation

File: docs/gomc.rst [hunk 2]
```diff
@@ -0,0 +1,32 @@
+.. _gomc-datasets:
+
+===============
+GOMC datasets
+===============
+.. automodule:: alchemtest.gomc
+
+The :mod:`alchemlyb.gomc` module features datasets generated using the
```
Comment (orbeckst) [inline docs/gomc.rst]: Can you briefly explain what "GOMC" stands for and what kind of approach it is?

File: src/alchemtest/gomc/access.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    module_path = dirname(__file__)
+
+    data = sorted(glob(join(module_path, 'benzene', 'inWater', 'Free_Energy_BOX_0_PRODUCTION_*.dat')))
+
+    with open(join(module_path, 'benzene', 'descr.rst')) as rst_file:
+        fdescr = rst_file.read()
+
+    return Bunch(data=data,
+                 DESCR=fdescr)
```
Comment (orbeckst) [inline src/alchemtest/gomc/access.py]: possibly add a newline here

File: src/alchemtest/gomc/benzene/descr.rst [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+Notes
+-----
+Data Set Characteristics:
+    :Number of Legs: 2 (Coulomb, VDW)
+    :Number of Windows: 7 for Coulomb, 15 for VDW
+    :Length of Windows: 50 million Monte Carlo steps
+    :System Size: 1001 molecules
+    :Temperature: 298 K
+    :Pressure: 1 bar
```
Comment (orbeckst) [inline src/alchemtest/gomc/benzene/descr.rst]: Please also add the units for energy and time, similar to https://github.com/alchemistry/alchemtest/blob/master/src/alchemtest/gmx/benzene/descr.rst (was recently added in PR #32 ).

Comment (orbeckst) [reply src/alchemtest/gomc/benzene/descr.rst]: And yes, the unit of time can just be "MC step".

File: docs/gomc.rst (hunk 2 above)
Comment (msoroush) [reply docs/gomc.rst]: GOMC stands for GPU Optimized Monte Carlo. GOMC is open-source software for simulating molecular systems using the Metropolis Monte Carlo algorithm.

Would you like me to add this information to this file?

**Comment by msoroush** [comment]
> This is already looking very good. Some minor text comments.
> 
> Can you try building the docs locally with
> 
> ```
> python setup.py build_sphinx
> ```
> 
> and then browse the files starting at `build/sphinx/html/index.htm`. Just check that your docs render the way you expect them to look like and let us know.
> 
> P.S.: Apologies for the delay in reviewing. Feel free to ping me with @orbeckst when you haven't gotten a reply within a few days.

No worries, I know you are very busy. I already checked the rendering before pull request. I rebuilt the document after applying the changes that you asked. and very thing looks normal.

File: docs/gomc.rst (hunk 2 above)
Comment (msoroush) [reply docs/gomc.rst]: > Can you briefly explain what "GOMC" stands for?

I added "GOMC stands for GPU Optimized Monte Carlo."

> what kind of approach it is?

Would you like me explain the simulation parameters (Monte Carlo moves and their frequencies) or you are referring to something else?

Comment (orbeckst) [reply docs/gomc.rst]: Details can be added to the data set itself if you think it would be useful. I am happy with the changes you made.

**Comment by orbeckst** [comment]
I am merging this. If you want to add more then please open a new PR. 

Congratulations (and thank you) for your first merged PR!

**Pull request merged**